core/src/main/java/io/questdb/griffin/engine/table/ShowPartitionsRecordCursorFactory.java

```diff
@@ -0,0 +1,434 @@
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.table;
+
+import io.questdb.cairo.*;
+import io.questdb.cairo.sql.Record;
+import io.questdb.cairo.sql.RecordCursor;
+import io.questdb.cairo.sql.RecordMetadata;
+import io.questdb.griffin.PlanSink;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.str.SizePrettyFunctionFactory;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.std.*;
+import io.questdb.std.str.Path;
+import io.questdb.std.str.StringSink;
+
+import java.io.Closeable;
+
+public class ShowPartitionsRecordCursorFactory extends AbstractRecordCursorFactory {
+
+    private static final Log LOG = LogFactory.getLog(ShowPartitionsRecordCursor.class);
+    private static final RecordMetadata METADATA;
+
+    private final ShowPartitionsRecordCursor cursor = new ShowPartitionsRecordCursor();
+    private final TableToken tableToken;
+
+    public ShowPartitionsRecordCursorFactory(TableToken tableToken) {
+        super(METADATA);
+        this.tableToken = tableToken;
+    }
+
+    @Override
+    public RecordCursor getCursor(SqlExecutionContext executionContext) {
+        return cursor.of(executionContext, tableToken);
+    }
+
+    @Override
+    public boolean recordCursorSupportsRandomAccess() {
+        return false;
+    }
+
+    @Override
+    public void toPlan(PlanSink sink) {
+        sink.type("show_partitions").meta("of").val(tableToken);
+    }
+
+    private static class ShowPartitionsRecordCursor implements RecordCursor {
+        private final ObjList<String> attachablePartitions = new ObjList<>(4);
+        private final ObjList<String> detachedPartitions = new ObjList<>(8);
+        private CairoConfiguration cairoConfig;
+        private int dynamicPartitionIndex = -1;
+        private int limit; // partitionCount + detached + attachable
+        private Path path;
+        private int partitionBy = -1;
+        private int partitionIndex = -1;
+        private final PartitionsRecord partitionRecord = new PartitionsRecord();
+        private int rootLen;
+        private final StringSink sink = new StringSink();
+        private TableToken tableToken;
+        private final TxReader tableTxReader = new TxReader(FilesFacadeImpl.INSTANCE);
+        private CharSequence tsColName;
+
+        @Override
+        public void close() {
+            Misc.free(tableTxReader);
```

Comment:
there is no test that runs cursor more than once? final object closed while cursor is reusable ?